### ISSUE TITLE
Error using OMD-30-SE-100N sensors

### ISSUE THREAD

**Issue by milowangjinhong** [alknemeyer/optoforce#1]
Hi,

I was trying to have real-time readings of the OptoForce sensors and I am glad that I found this project! Thank you very much. 

According to the datasheet, I think I have Multi-channel DAQ v5.3 with three OMD-30-SE-100N sensors and hence imported OptoForce34. However, when I was trying to print the readings, the following error was raised, and I did not manage to find a way to solve it.

serial.serialutil.SerialException: GetOverlappedResult failed (OSError(22, 'Overlapped I/O event is not in a signaled state.', None, 996))

I am wondering if you know what is going on here and if you have any suggestions. Thank you very much!
![1](https://user-images.githubusercontent.com/53256896/106483313-4a5eaf00-64a6-11eb-94be-2a69a379947d.png)

**Comment by alknemeyer** [comment]
Hey,

I'm very happy to hear that it could be of use to you!

In terms of your bug: the error is happening in the PySerial library, which this package depends on.

Based on the line `Original message: PermissionError(...)` it seems that Python doesn't the required permissions. That apparently happens when [something else is using the device](https://stackoverflow.com/a/63922626/1892669). In that case, making sure no other program is connected to the device should work. You could do that by eg. unplugging the sensor

Otherwise it could be useful to go through my [blog post](https://alknemeyer.github.io/embedded-comms-with-python/) where I describe how this module was made. I unfortunately don't run windows and can't recreate the bug on my side :/

Let me know how it goes, so I can update the documentation in any case

**Comment by milowangjinhong** [comment]
Thank you for your response, and apologies for the delay as I was held by another project this week.

As for the posted bug, you are very right about it, thank you very much. In addition, I found some problems using the library and have fixed them. I am quite new to GitHub so I am not sure how I can contribute to the code directly so I'll list them down here for your reference:

- For the version of python I am using (3.6.12), and the Literal function is not available and I had to install the [typing-extensions](https://pypi.org/project/typing-extensions/) (3.7.4.3) and import the Literal from there: from typing_extensions import Literal.

- Since I am using multichannel DAQ, when I try to run your example code, I got stuck in the opt_ser.read_until(expected_header) line and I found that it is because the difference in the expected headers, shown in the figure below. Therefore, I had to update that value to expected_header = bytes((170, 7, 8, 28)) from the original bytes((170, 7, 8, 10))
![image](https://user-images.githubusercontent.com/53256896/107124889-3946e100-689e-11eb-8d86-246ac5600874.png)

-  Finally, read_fz = lambda: force_sensor.read(only_latest_data=False).Fz didn't quite work out for me as the name of the data was Fz1, Fz2 and Fz3 instead of Fz since I have more channels with my Daq. Although easy to correct, maybe worth to include that in the documentation to avoid confusion?

Again, thanks a lot for your work, now the code works perfectly well and will be very useful for my project. Please me know if anything above is not clear and I can try to help with it. :)

**Comment by alknemeyer** [comment]
I'm glad to hear that the fix worked.

Thanks for letting me know about the bugs. I've fixed them but if you would like to work on the library in the future, I'm happy to explain the process. In short, you'd need to `git clone` the library, (ideally) create a new branch (`git checkout -b the-name-of-your-feature`), modify the files, increment the version number (as explained near the end of the readme), `git add`, `git commit` and `git push` the changes, and then ask me to publish the new version to PyPI. It may seem like a lot but it's actually quite simple once you've done it. Don't worry about messing up :)

Anyway:

- I've added a conditional dependency on `typing-extensions` for Python < v3.8
- The headers have been updated - silly of me to miss that!
- I've clarified that the packets are different for each sensor, in the readme

I also added some stuff for error checking -- let me know if you end up using that

Good luck for your project!
